src/libraries/Common/src/System/HexConverter.cs

```diff
@@ -108,40 +109,55 @@ private static void EncodeToUtf16_Ssse3(ReadOnlySpan<byte> bytes, Span<char> cha
                                  (byte)'8', (byte)'9', (byte)'a', (byte)'b',
                                  (byte)'c', (byte)'d', (byte)'e', (byte)'f');
 
+
+            nuint lengthSubVector128 = (nuint)bytes.Length - (nuint)Vector128<int>.Count;
             do
             {
                 // Read 32bits from "bytes" span at "pos" offset
                 uint block = Unsafe.ReadUnaligned<uint>(
                     ref Unsafe.Add(ref MemoryMarshal.GetReference(bytes), pos));
 
+                // TODO: Remove once cross-platform Shuffle is landed
+                // https://github.com/dotnet/runtime/issues/63331
+                [MethodImpl(MethodImplOptions.AggressiveInlining)]
+                static Vector128<byte> Shuffle(Vector128<byte> value, Vector128<byte> mask) =>
+                    Ssse3.IsSupported ? Ssse3.Shuffle(value, mask) : AdvSimd.Arm64.VectorTableLookup(value, mask);
+
                 // Calculate nibbles
-                Vector128<byte> lowNibbles = Ssse3.Shuffle(
+                Vector128<byte> lowNibbles = Shuffle(
                     Vector128.CreateScalarUnsafe(block).AsByte(), shuffleMask);
-                Vector128<byte> highNibbles = Sse2.ShiftRightLogical(
-                    Sse2.ShiftRightLogical128BitLane(lowNibbles, 2).AsInt32(), 4).AsByte();
+
+                // ExtractVector128 is not entirely the same as ShiftRightLogical128BitLane, but it works here since
+                // first two bytes in lowNibbles are guaranteed to be zeros
+                Vector128<byte> shifted = Sse2.IsSupported ?
+                    Sse2.ShiftRightLogical128BitLane(lowNibbles, 2) :
+                    AdvSimd.ExtractVector128(lowNibbles, lowNibbles, 2);
```

Comment:
Is this better than using `Zero` for the upper?

Reply:
> Is this better than using `Zero` for the upper?

Not sure I understand, this code only shifts values by 2 elements (byte-wide) right

Reply:
Nevermind. I was thinking of the operation incorrectly and thinking we had an instruction which just shifted the elements over one
but shift right and insert is still per element.